src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -629,7 +629,9 @@ private void visitUnionType(UnionType ut) {
         public boolean apply(JSType input) {
           // Skip - TypeScript does not have explicit null or optional types.
           // Optional types must be handled at the declaration name (`foo?` syntax).
-          return !input.isNullable() && !input.isVoidType();
+          return !input.isNullable() && !input.isVoidType()
+              // Template types are always nullable, but we should not skip them
+              || input.isTemplateType();
```

Comment:
The intention of the code was that unions commonly contain `Foo|null`, and we'd like to filter out the `null` case. Maybe the Real Bug is that this should not be `!input.isNullable()` but rather `!input.isSpecialNullType()` (I'm making that up)? I.e. the difference between the type permitting/including `null`, and the type actually being the special `null` type?